### PR TITLE
Add migrate task before imports

### PR DESCRIPTION
**Pull request by roel-devries**
Migrate task, as part of initialise DAG, is inserted once before all  import tasks. 
The last node(s) of prepare is linked to this DAG if available.

### CODE REVIEW

File: dags/benk/generate.py [hunk 1]
```diff
@@ -31,6 +33,9 @@
                 Relate(definition.catalog, collection.collection, relation) for relation in collection.relations
             ]
 
+        # Start with initialising / migrating before imports
+        chain(*initialise.get_leaf_nodes(), flatten_list([i.get_start_nodes() for i in imports]))
```
Comment (JKoetsier) [inline dags/benk/generate.py]: This doesn't seem right. You're unpacking the leaf nodes, so all leaf nodes will be chained one after the other. What you want is start the imports after all leaf nodes have ended separately.

Say initialise has two leaf nodes, L1 and L2, with import i1, i2 and i3. What you do now is:

L1 -> L2 -> [i1, i2, i3], or written out:

L1 -> L2
L2 -> i1
L2 -> i2
L2 -> i3

What you want is [L1, L2] -> [i1, i2, i3], or, written out:

L1 -> i1
L1 -> i2
L1 -> i3
L2 -> i1
L2 -> i2
L2 -> i3

Of course, when there is only 1 leaf node, it doesn't really matter in this case. See what happens when you leave out the rules starting with L1 above. The result is the same. It goes wrong when there are multiple leaf nodes, and it's always possible that a new leaf node will be added in the future. Then the script will break.

Instead, use
`cross_downstream(initialise.get_leaf_nodes(), flatten_list([i.get_start_nodes() for i in imports]))`

File: dags/benk/generate.py [hunk 2]
```diff
@@ -40,4 +45,4 @@
             prepare = Prepare(definition.catalog)
 
             # Add Prepare
-            cross_downstream(prepare.get_leaf_nodes(), flatten_list([i.get_start_nodes() for i in imports]))
+            chain(prepare.get_leaf_nodes(), *initialise.get_start_nodes())
```
Comment (JKoetsier) [inline dags/benk/generate.py]: Same here, you're chaining all possible start nodes, while they should be on the same level.

Say you have prepare leaf nodes L1, L2, and initialise start nodes i1 and i2.
What you do now:

[L1, L2] -> i1 -> i2
The start nodes are depending on each other here, while they should start at the same time: [L1, L2] -> [i1, i2]

Again, no big issue when here is only one start node in initialise (which is the case at this moment), but it doesn't agree with the get_start_nodes() signature.

`cross_downstream(prepare.get_leaf_nodes(), initialise.get_start_nodes())`

**Review by JKoetsier — CHANGES_REQUESTED** [review]

**Review by JKoetsier — APPROVED** [review]

File: dags/benk/generate.py (hunk 1 above)
Comment (roel-devries) [reply dags/benk/generate.py]: fixed

File: dags/benk/generate.py (hunk 2 above)
Comment (roel-devries) [reply dags/benk/generate.py]: fixed

**Pull request merged**